applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp

```diff
@@ -394,6 +395,99 @@ void BaseSolidElement::CalculateRightHandSide(
     CalculateAll( temp, rRightHandSideVector, rCurrentProcessInfo, CalculateStiffnessMatrixFlag, CalculateResidualVectorFlag );
 }
 
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void BaseSolidElement::AddExplicitContribution(
+    const VectorType& rRHSVector, 
+    const Variable<VectorType>& rRHSVariable,
+    Variable<double>& rDestinationVariable,
+    const ProcessInfo& rCurrentProcessInfo
+    ) 
+{
+    KRATOS_TRY;
+
+    auto& r_geom = this->GetGeometry();
+    const SizeType dimension = r_geom.WorkingSpaceDimension();
+    const SizeType number_of_nodes = r_geom.size();
+    const SizeType element_size = dimension * number_of_nodes;
+
+    // Compiting the nodal mass
+    if (rDestinationVariable == NODAL_MASS ) {
+        Matrix element_mass_matrix = ZeroMatrix(element_size, element_size);
+        ProcessInfo temp_process_information = rCurrentProcessInfo; // Dummy
+        temp_process_information.SetValue(COMPUTE_LUMPED_MASS_MATRIX, true);
+        this->CalculateMassMatrix(element_mass_matrix, temp_process_information);
+
+        for (IndexType i = 0; i < number_of_nodes; ++i) {
+            double aux_nodal_mass = 0.0;
+
+            const IndexType index = i * dimension;
+
+            for (IndexType j = 0; j < element_size; ++j) {
+                aux_nodal_mass += element_mass_matrix(index, j);
+            }
+
+            #pragma omp atomic
```

Comment:
atomic is better than `SetLock` and `UnSetLock` right?

Reply:
Yes, much more